integration/spark/src/main/scala/org/apache/spark/sql/parser/CarbonSparkSqlParserUtil.scala

```diff
@@ -447,21 +447,31 @@ object CarbonSparkSqlParserUtil {
       table: String,
       columnName: String,
       columnNameCopy: String,
-      dataType: String,
+      dataType: Option[String],
       values: Option[List[(Int, Int)]],
-      comment: Option[String]
+      comment: Option[String],
+      complexChild: Option[Field]
   ): CarbonAlterTableColRenameDataTypeChangeCommand = {
     val isColumnRename = !columnName.equalsIgnoreCase(columnNameCopy)
-    val alterTableColRenameAndDataTypeChangeModel =
+    val alterTableColRenameAndDataTypeChangeModel = if (!dataType.equals(None)) {
```

Comment:
instead of duplicating the whole thing, just put the columnName/columnNameCopy and datatype.get.tolowerCase/complexChild to 2 variables depending on the conditions and use them, because most of the variables are same.

Reply:
done